### PR TITLE
feat: validate customization pre commit config

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
-Validate customization pre-commit config. Issue - https://github.com/agritheory/test_utils/issues/1
checked `validate_customizations` from the repos `inventory_tools`, `beam`, `cloud_storage`, `check_run`, `us_regional`, `electronic_payments` and generalized validate_customizations.

### CODE REVIEW

**Comment by agritheory** [comment]
@MyuddinKhatri I have two more that we need to reconcile as well, I'll send them privately since they're in customer repo but don't contain any PIA.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -42,6 +42,15 @@ repos:
         language: system
         types: [python]
 
+  - repo: local
```
Comment (agritheory) [inline .pre-commit-config.yaml:45]: We don't actually want this to run as local, but instead against the repo. Try replacing the pre-commit in Check Run or another target repo with this one and refactor the addresses to work with the remote, more or less like this (which I'm sure is incorrect)

```yaml
  - repo: https://github.com/agritheory/test_utils
    rev: v2.2.6
    hooks:
      - id: validate_javascript_dependencies
        always_run: true
        name: pre-commit/validate_javascript_dependencies.py
        entry: python3 pre-commit/validate_javascript_dependencies.py
        language: system
        types: [python]
      - id: validate_python_dependencies
        always_run: true
        name: pre-commit/validate_python_dependencies.py
        entry: python3 pre-commit/validate_python_dependencies.py
        language: system
        types: [python]
      - id: validate_customizations
        always_run: true
        name: pre-commit/validate_customizations.py
        entry: python3 pre-commit/validate_customizations.py
        language: system
        types: [python]
```

**Review by agritheory — CHANGES_REQUESTED** [review]
Let's refactor this  to be closer to the as-used use case.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (agritheory) [reply .pre-commit-config.yaml:45]: The rev in that note is also wrong

Comment (MyuddinKhatri) [reply .pre-commit-config.yaml:45]: > We don't actually want this to run as local, but instead against the repo. Try replacing the pre-commit in Check Run or another target repo with this one and refactor the addresses to work with the remote, more or less like this (which I'm sure is incorrect)
> 
> ```yaml
>   - repo: https://github.com/agritheory/test_utils
>     rev: v2.2.6
>     hooks:
>       - id: validate_javascript_dependencies
>         always_run: true
>         name: pre-commit/validate_javascript_dependencies.py
>         entry: python3 pre-commit/validate_javascript_dependencies.py
>         language: system
>         types: [python]
>       - id: validate_python_dependencies
>         always_run: true
>         name: pre-commit/validate_python_dependencies.py
>         entry: python3 pre-commit/validate_python_dependencies.py
>         language: system
>         types: [python]
>       - id: validate_customizations
>         always_run: true
>         name: pre-commit/validate_customizations.py
>         entry: python3 pre-commit/validate_customizations.py
>         language: system
>         types: [python]
> ```

Yes, I had a plan once we complete all the pre-commit configurations then we can start putting this as a pre-commit hook in other repos. I'll make the change.

Comment (agritheory) [reply .pre-commit-config.yaml:45]: @MyuddinKhatri OK that makes sense. Let's do that after the rest of the pre-commit scripts are done.

**Pull request merged**